### PR TITLE
Add container version to health check

### LINKED ISSUE #507: Add container version number to /health-check endpoint
Per discussion in 5/25 ops/dev meeting, we should have an easy way to check what version of the project is being run by ECS as a manual way to test an update. I propose adding this information to the `/health-check` endpoint as an additional `"version"` field.

This would also be a good first step toward building a more robust monitoring solution, as we could add information about database, redis, etc. to the response object.

### PR DESCRIPTION
**Pull request by jonseitz**
This PR adds an additional field to our `/health-check` endpoint to display the version number of the container image that's actively running. This was a request from the Ops team to help verify that redeploying the app on ECS actually launches the new code as expected.

I originally approached this by adding an environment variable that would be set by a build argument in Docker. That made it easier to test, as the environment variable could simply be overwritten at run time. However, that's also a downside as it removes some of the confidence that the version number is accurate.

Instead of setting the build argument as a environment variable, we're writing it to a `.dockerversion` file in the application root. This creates an additional layer in our build process, which ensures that the final SHA hash of the container will change any time the version changes and give us a little more confidence in the accuracy of the value. (The file could theoretically be overwritten at runtime, but 1) the `node` user that runs the container process only has read access to the file and 2) our ECS configuration should be pulling a fresh container image down, so any changes to the file would be replaced with the layer in the registry.

The build argument will be set by the GitHub Actions workflow that publishes the container image, which does make this slightly harder to manually test. As was the case with the previous PR related to the workflows (#489) the updated workflow file needs to be on the `main` branch before it takes effect. My best recommendation for testing is to fork the repo, merge this branch into main, then run the container it publishes and check the health-check output. For mine, I did this by running:

```
docker-compose up -d
docker-compose stop node
docker run -it --rm --env-file .env --network course-planner_default -p 127.0.0.1:3001:3001 ghcr.io/jonseitz/course-planner-action-test:stable
curl localhost:3001/course-planner/health-check
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality
      to change)

## Checklist:

- [x] I have run `eslint` on the code
- [x] I have added JSDoc for all of my code (where applicable)
- [x] I have added tests to cover my changes.

## Priority:
- [x] Normal 
- [ ] High 

## Related Issues:

Fixes #507

### CODE REVIEW

**Review by rmainseas — CHANGES_REQUESTED** [review]
This is looking great. Just a quick thing about the `fs.*` calls in the config service. I'm currently running into an error from the semester service in testing, but I'm not going to hold this review up for that so I'll put this back to you while I try to deal with that...

File: src/server/config/config.service.ts [hunk 1]
```diff
@@ -314,6 +316,25 @@ class ConfigService {
...
   }
+
+  /**
+   * Attempt to read the current build version from the .dockerversion file, or
+   * else return an empty string.
+   */
+  public get buildVersion(): string {
+    // The file should be created when the image is built
+    const versionFile = path.resolve('.dockerversion');
+    if (fs.existsSync(versionFile)) {
```
[the remark this answers is not in the document]
Comment (jonseitz) [reply src/server/config/config.service.ts:332]: Ah, that's a great point, especially as the `/health-check` endpoint is going to be called frequently by the load balancer in production.

I didn't really see how that would be added to the `ConfigModule` constructor though, so I changed some things around in 188e8311 to read it in the `ConfigService` constructor instead.

I also added a small change to use `.trim()` in place of the `.replace()` call I had before. That better accomplished what I wanted to do and doesn't have the overhead of compiling the regex on every call.

**Review by rmainseas — APPROVED** [review]
LGTM. I've made  suggestion below about how one might go about computing that value in the config module and passing it to the config class. However, I'll leave it to your judgement as to whether or not you want to go ahead with that. 

I should also note that while I did test the code chunk below, I only tested the happy path so left out error handling, null checks etc. for brevity as a proof of concept.
